src/pages/SignUp/SignUp.js

```diff
@@ -1,7 +1,276 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import './SignUp.scss';
 
 const SignUp = () => {
-  return <div>SignUp</div>;
+  const navigate = useNavigate();
+  const [inputValue, setInputValue] = useState({
+    email: '',
+    password: '',
+    passwordAgain: '',
+    name: '',
+    phoneNumber: '',
+    birthday: '',
+    adress: '',
+  });
+  const {
+    email,
+    password,
+    passwordAgain,
+    name,
+    phoneNumber,
+    birthday,
+    adress,
+  } = inputValue;
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setInputValue({ ...inputValue, [name]: value });
+  };
+
+  const goToMain = () => {
+    if (
+      emailCheck &&
+      passwordCheck &&
+      passwordCheckAgain &&
+      namecheck &&
+      phoneNumberCheak &&
+      birthdayCheck &&
+      adressCheck &&
+      isAllChecked
+    ) {
+      navigate('/');
+    } else {
+      alert('필수 정보를 정확하게 입력 해주세요');
+    }
+  };
+
+  const emailCheck =
+    !email || (email.includes('@', 5) && email.includes('.com', 9));
+  const passwordCheck =
+    !password ||
+    new RegExp(/^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})/).test(
+      password
+    );
+  const passwordCheckAgain = !passwordAgain || password === passwordAgain;
+  const namecheck = name.length >= 2;
+  const phoneNumberCheak = phoneNumber.length >= 10 && phoneNumber.length <= 12;
+  const birthdayCheck = birthday.length === 6;
+  const adressCheck = adress.length > 8;
+
+  const [isButtonWork, setIsButtonWork] = useState({
+    agree: false,
+    userInfo: false,
+    age: false,
+  });
+  const { agree, userInfo, age } = isButtonWork;
+
+  const isAllChecked = Object.values(isButtonWork).every(
+    value => value === true
+  );
+
+  const handleCheckBox = e => {
+    setIsButtonWork(prev => ({
+      ...prev,
+      [e.target.name]: !prev[e.target.name],
+    }));
+  };
+
+  const handleAllCheck = () => {
+    const checkboxKeys = Object.keys(isButtonWork);
+    checkboxKeys.forEach(list => {
+      return setIsButtonWork(prev => ({ ...prev, [list]: !isAllChecked }));
+    });
+  };
+
+  return (
+    <div className="container">
```

Comment:
```suggestion
    <div className="signup">
```
scss 사용할 때 가장 부모 태그의 classname은 해당 컴포넌트의 이름과 동일하게 해주세요!

Reply:
네 확인하고 수정했습니다 :)